File: MineInAbyss-API/src/main/java/com/derongan/minecraft/mineinabyss/World/AbyssWorldManager.java
package com.derongan.minecraft.mineinabyss.World;

import java.util.List;

/**
 * Manages layers and sections
 */
public interface AbyssWorldManager {
    /**
     * Get all layers in index order.
     * @return All layers
     */
    List<Layer> getLayers();

    /**
     * Get all sections in index order
     * @return All sections
     */
    List<Section> getSections();

    /**
     * Get the section at index or null if out of bounds
     * @param index the index
     * @return the section to get
     */
    Section getSectonAt(int index);

    /**
     * Get the layer at index or null if out of bounds
     * @param index the index
     * @return the layer to get
     */
    Layer getLayerAt(int index);

    /**
     * Check if the world is a world set up for the abyss
     * @param worldName
     * @return
     */
    boolean isAbyssWorld(String worldName);
}

File: src/main/java/com/derongan/minecraft/mineinabyss/services/AbyssWorldManager.kt
package com.derongan.minecraft.mineinabyss.services

import com.derongan.minecraft.deeperworld.world.section.Section
import com.derongan.minecraft.mineinabyss.world.Layer
import com.derongan.minecraft.mineinabyss.world.LayerKey
import com.mineinabyss.idofront.plugin.getService
import org.bukkit.World

/**
 * Manages layers and sections
 *
 * @property layers [Layer]s associated by their [LayerKey]

 */
interface AbyssWorldManager {
    companion object : AbyssWorldManager by getService()

    val layers: Map<LayerKey, Layer>

    /**
     * @param world
     * @return whether the world is a world set up for the abyss
     */
    fun isAbyssWorld(world: World): Boolean

    fun getLayerForSection(section: Section): Layer?

    fun getLayerFor(key: LayerKey): Layer?
}
